Vejleder rettelser (#14)
chhoumann/p5: sections/Chapter 5 Sprint 3/data-access.tex

\subsection{Object Relational Mapping}

> An Object Relational Mapping (ORM) is often used in an application to connect to, and manipulate data on a database.

Reviewer: ```suggestion
An Object Relational Mapping (ORM) is often used in an application to connect to and manipulate data on a database.
```

> The primary requirement for the choice of an ORM was fast implementation and ease of use, due to the timeline constraints that were imposed by the \knox{} project. 

Reviewer: ```suggestion
The primary requirement for the choice of an ORM was fast implementation and ease of use due to the timeline constraints that were imposed by the \knox{} project. 
```

\subsubsection{Dapper}

> Dapper is an open source Micro ORM. This means that Dapper is a mapping framework that helps mapping the result from a native query to a class with the same attributes as the tables that are being queried. Dapper is a light weight framework made for developers that prefers to use stored procedures or native query language over using a large scale ORM tool. 

Reviewer: ```suggestion
Dapper is an open source Micro ORM. It is a mapping framework that helps mapping the result from a native query to a class with the same attributes as the tables. Dapper is a lightweight framework made for developers that emphasizes using stored procedures or native query language over using a large scale ORM tool. 
```

> Dapper, when added to projects, extends the IDbConnection inferface. This extension adds helpers. 

Reviewer: ```suggestion
Dapper, when added to projects, extends the \texttt{IDbConnection} inferface. This extension adds helper methods. 
```

> One of the helpers is the Query() method. The query method will take an sql query as a parameter, the method returns a list. The type of list that is returned can either be a strongly typed list or a list of dynamic objects.

Reviewer: ```suggestion
One of the helpers is the \texttt{Query} method. This method will take an SQL query as a parameter and return a list. The returned list can either be strongly or dynamically typed.
```

> Another helper that is added to IDbConnection is the Execute() method which takes some SQL commands as a parameter, and returns no results. When the method is called the commands used as parameter will be executed. The Execute() method can also be used to execute commands multiple times\cite{Dapper_Git}.

Reviewer 2: ```suggestion
Another helper that is added to \texttt{IDbConnection} is the \texttt{Execute} method. This method takes one or more SQL commands as a parameter and has no return type. When called, the commands passed to the method will be executed. The \texttt{Execute} method can also be used to execute commands multiple times\cite{Dapper_Git}.
```

Dappers simple approach to ORM's means that a large part of the standard features have been dropped. This means that Dappers focuses on being a lightweight and efficient framework that should cover most of the users needs over being a fullfledge ORM \cite{Dapper_Git}.

This means that Dapper as a framework covers our usecases with the caveat that additional time is going to be needed in order to fullfil the sprints MVP. 

\subsubsection{Entity Framework Core}

> The second technology that we considered was Entity Framework Core (EF Core), which is a data access technology developed by Microsoft for the .NET platform. EF Core can serve as an ORM \cite{Object_relational_mapping} which lets developers abstract away the complications of converting data to .NET objects, in addition to eliminating the majority of the data-access code that would typically have to be written.

Reviewer 2: ```suggestion
The second technology that we considered was Entity Framework Core (EF Core), which is a data access technology developed by Microsoft for the .NET platform. EF Core can serve as an ORM \cite{Object_relational_mapping}, which lets developers abstract away the complications of converting data to .NET objects and eliminates the majority of the data-access code that would typically have to be written.
```
